spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/SparkV2Filters.java

```diff
@@ -245,6 +251,46 @@ public static Expression convert(Predicate predicate) {
     return null;
   }
 
+  private static <T> PredicateChildren<T> predicateChildren(Predicate predicate) {
+    T value;
+    UnboundTerm<T> term;
+    if (isRef(leftChild(predicate)) && isLiteral(rightChild(predicate))) {
+      term = ref(SparkUtil.toColumnName(leftChild(predicate)));
+      value = convertLiteral(rightChild(predicate));
+    } else if (isRef(rightChild(predicate)) && isLiteral(leftChild(predicate))) {
+      term = ref(SparkUtil.toColumnName(rightChild(predicate)));
+      value = convertLiteral(leftChild(predicate));
+    } else {
+      return null;
+    }
+
+    return new PredicateChildren<>(term, value);
+  }
+
+  private static <T> UnboundPredicate<T> handleEqual(PredicateChildren<T> children) {
```

Comment:
Can you move this back to where it was so that the diff is readable?

Reply:
Updated.